app.py

```diff
@@ -303,12 +295,9 @@ def setup_language():
     g.keyword_lang = session['keyword_lang']
 
     # Set the page direction -> automatically set it to "left-to-right"
-    # Switch to "right-to-left" if one of the language in the list is selected
+    # Switch to "right-to-left" if one of the language is rtl according to Locale (from Babel) settings. 
     # This is the only place to expand / shrink the list of RTL languages -> front-end is fixed based on this value
-    g.dir = "ltr"
-    if g.lang in ['ar', 'he', 'ur']:
-        g.dir = "rtl"
-
+    g.dir = Locale(g.lang).text_direction
```

Comment:
What if, for some reason the `Locale` function fails? It is nice to always have a "default" value for this g attribute as it is crucial for our front-end. I propose to change this to:

```
g.dir = "ltr"
g.dir = Locale(g.lang).text_direction
```

Reply:
On seconds thoughts, this will still result in the same error if the `Locale` function doesn't return anything. So the best solution might be something like this:

```
g.dir = "ltr"
if Locale(g.lang).text_direction in ["ltr", "rtl"]:
    g.dir = Locale(g.lang).text_direction

Reply:
That's very defensive and calling the Locale getter twice, but yeah, it's save.